### ISSUE TITLE
How to debug custon site?

### ISSUE THREAD

**Issue by letterale** [dobbelina/repository.dobbelina#967]
Hi,
I am trying to add a custom site but, once installed and enabled, I receive error via popup window that warns me that it will be disabled.

I guess there is a debug option somewhere but I still don't understand where to look for it. 
Ask: there is an option to add (in the Custom site code or in some other file) to understand what has gone wrong, what kind of error it is or what line of the Custom site code generates the error.

Can someone help me?

**Comment by DJMix-I-Can** [comment]
The Debug file is within Kodi itself. You need to go to settings, system files, system, logging and check off enable debug logging in order to enable it. Once it's enabled, replicate the installation of that custom site you're trying to install. When you get the error pop up, you can disable the debug log & the proceed to upload it. From the sounds of it, it sounds like you're trying to learn. That is good, you're taking the initiative to learn. The kodi log file sometimes gives you the errors you're encountering, without enabling the debug setting. The question right now is what device are you using? Android, Windows etc... The path to the log files are different depending on the OS. (Operating System) If you're willing to share that custom site in order to see the coding, it could also be helpful. I myself have also been trying to learn this coding & the brains that have been keeping this addon up to date have been very helpful. As long as you're willing to learn.....

**Comment by letterale** [comment]
Thank you for your support and yes, obviously i'm a curius newbbie who trying to learn.

The site is eroprofile.com, the error is invalid syntax (custom_1.py, line 45), look file attach.
[eroprofile.zip](https://github.com/dobbelina/repository.dobbelina/files/10709713/eroprofile.zip)

Now i still work with window but if this custom site work right, i would like instal in nvidia shield also.

**Comment by DJMix-I-Can** [comment]
On Windows you could look at the log file via C:\Users\YOURUSERNAME\AppData\Roaming\Kodi
You will need to have "show hidden files, folders & drives" enabled in order to see the AppData folder.
On Android, you could find it at Android/data/org.xbmc.kodi/files/.kodi/temp folder.

**Comment by Gujal00** [comment]
@letterale  
> The site is eroprofile.com, the error is invalid syntax (custom_1.py, line 45), look file attach.
Line 44 is the problem it is truncated, missing code at the end
currently it is `match = re.compile(r'bg-black"><a href="([^"]+).+?<div class="videoTtl" title="([^"]+)', re.DOTALL`
change it to
```python
match = re.compile(r'bg-black"><a href="([^"]+).+?<div class="videoTtl" title="([^"]+)', re.DOTALL | re.IGNORECASE).findall(listhtml)
```

**Comment by dobbelina** [comment]
@Gujal00 

Thanks for this.
I have added this and other custom sites that i have collected together
under this topic [Cumination - "Custom Sites" How does it work?](https://github.com/dobbelina/repository.dobbelina/issues/407) that i have also pinned.

**Comment by letterale** [comment]
thank you all, I only passed the URL and the video title just to simplify this first attempt, if I understand what I do, I will try with other elements.

**Comment by letterale** [comment]
et voilà, added category and search. now its fine ;-)
[custom-eroprofile.zip](https://github.com/dobbelina/repository.dobbelina/files/10713324/custom-eroprofile.zip)

**Comment by DJMix-I-Can** [comment]
I have something for you to check out. When you select a category & hit next page, the first page reloads. It also happens after you do a search. The next page won't load. It would still be the first page.

**Comment by ErosVece** [comment]
I love this, this is one of main reasons why custom sites was added. To give others an option or something to experiment with for adding their favorite sites to Cumination :)

**Comment by letterale** [comment]
> I have something for you to check out. When you select a category & hit next page, the first page reloads. It also happens after you do a search. The next page won't load. It would still be the first page.

You're right, too much enthusiasm did not make me check well. Now I have changed it, even the pagination in the categories and in the search it seems to me ok.

If someone can confirm, then I try to update the zip also in the other discussion

[eroprofile.zip](https://github.com/dobbelina/repository.dobbelina/files/10715799/eroprofile.zip)

**Comment by DJMix-I-Can** [comment]
I hear you, I've been there. It's working now. Thanks for the share...

**Comment by TapeMeOh** [comment]
Is there a good guide or something on how to create a custom site?

Or some place where people can share ones they have created?

**Comment by Gujal00** [comment]
@TapeMeOh 
if you know python, then you can easily replicate using the various examples [on this pinned thread](https://github.com/dobbelina/repository.dobbelina/issues/407)
